vcr/verifier/verifier.go

```diff
@@ -145,3 +144,58 @@ func (v verifier) Verify(credentialToVerify vc.VerifiableCredential, allowUntrus
 
 	return nil
 }
+
+func (v *verifier) IsRevoked(credentialID ssi.URI) (bool, error) {
+	_, err := v.store.GetRevocation(credentialID)
+	if err != nil {
+		if errors.Is(err, ErrNotFound) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (v *verifier) CheckAndStoreRevocation(document proof.SignedDocument) error {
+	asBytes, err := json.Marshal(document)
+	revocation := credential.Revocation{}
+	if err := json.Unmarshal(asBytes, &revocation); err != nil {
+		return err
+	}
+
+	if err := credential.ValidateRevocation(revocation); err != nil {
+		return err
+	}
+
+	// issuer must be the same as vc issuer
+	subject := revocation.Subject
+	subject.Fragment = ""
+	if subject != revocation.Issuer {
+		return errors.New("issuer of revocation is not the same as issuer of credential")
+	}
+
+	vmWithoutFragment := revocation.Proof.VerificationMethod
+	vmWithoutFragment.Fragment = ""
+	if vmWithoutFragment != revocation.Issuer {
+		return errors.New("verificationMethod should owned by the issuer")
+	}
+
+	pk, err := v.keyResolver.ResolveSigningKey(revocation.Proof.VerificationMethod.String(), &revocation.Date)
+	if err != nil {
+		return fmt.Errorf("unable to resolve key for revocation: %w", err)
+	}
+
+	ldProof := proof.LDProof{}
+	if err := document.UnmarshalProofValue(&ldProof); err != nil {
+		return err
+	}
+	err = ldProof.Verify(document.DocumentWithoutProof(), signature.JSONWebSignature2020{ContextLoader: v.contextLoader}, pk)
+	if err != nil {
+		return fmt.Errorf("unable to verify revocation signature: %w", err)
+	}
+
+	if err := v.store.StoreRevocation(revocation); err != nil {
```

Comment:
Since `IsRevoked` is only supplied the credential ID (which can be whatever URI), can I (issuer `did:nuts:rein`) revoke you credentials if I specify IDs of credentials issued by you (issuer `did:nuts:steven`)? Shouldn't `IsRevoked` be also supplied with the issuer URI? Then you're safe, even when issuers accidentally (or on purpose) issue VCs with colliding IDs.

Reply:
The revocation validation step checks if the issuer of the revocation is also the issuer of the revoked credential. So, the answer should be no.